crates/fefix/src/session/connection.rs

```diff
@@ -213,102 +237,54 @@ where
     fn heartbeat(&self) -> Duration {
         self.config.heartbeat()
     }
-}
-
-pub struct MessageBuilder {}
-
-pub struct MessageBuiderTuple<'a> {
-    phantom: PhantomData<&'a ()>,
-}
-
-impl<'a> MessageBuiderTuple<'a> {
-    pub fn get(self) -> (EncoderHandle<'a, Vec<u8>>, &'a mut MessageBuilder) {
-        unimplemented!()
-    }
-}
-
-impl MessageBuilder {
-    pub fn start_message(&mut self, begin_string: &[u8], msg_type: &[u8]) -> MessageBuiderTuple {
-        unimplemented!()
-    }
-}
-
-struct ResponseData<'a> {
-    pub begin_stringt: &'a [u8],
-    pub msg_type: &'a [u8],
-    pub msg_seq_num: u32,
-}
-
-pub trait FixConnector<'a, B, C, Z>
```

Comment:
Removed having a `FixConnector` trait with default implementations because I quickly got into compilation issues when I needed to access attributes of the FixConnection struct - because of borrowing rules in rust 

For instance, when creating a new message I needed to call

1. self.encoder().start_message(..) - mutable call on self #1 to start encoding a message that
2. self.seq_numbers().get_incr_outbound(..) - mutable call on self #2 to increment the sequence numbers on the outbound method

I avoided this by placing all functionality in the FixConnection struct where I can call methods on the attributes directly benefiting from splitting borrows